### PR TITLE
Basic Go implementation

### PR DESCRIPTION
**Pull request by jackgene**
Initial stab at a Go implementation.

Does not use any concurrency library, just Go primitives (Go routines, channels) and `context.Context` from the standard library.

### CODE REVIEW

**Comment by jamesward** [comment]
Woah! Nice work Jack! This is great. One thing to discuss... I've tried in the other implementations not to have shared code between the scenarios. I was thinking that it is nice to be able to look at the full implementation for a given scenario without having to hop to some other place in the code.  I know there'd be a lot of duplication if that was enforced but it seems like that will help better portray what all it takes to accomplish each scenario individually.  WDYT?

**Comment by jackgene** [comment]
> Woah! Nice work Jack! This is great. One thing to discuss... I've tried in the other implementations not to have shared code between the scenarios. I was thinking that it is nice to be able to look at the full implementation for a given scenario without having to hop to some other place in the code. I know there'd be a lot of duplication if that was enforced but it seems like that will help better portray what all it takes to accomplish each scenario individually. WDYT?

Ah, that's right, I did remember you doing that in other places. I was originally thinking about making each scenario more easy to follow, but ended up with a lot of repeated code (that's Go for you).

So I just pushed a commit that gets us half way there - it shows how each scenario accomplishes what it needs to to fulfill each race's concurrency requirements, but without the HTTP error handling stuff making things too verbose and hard to follow.

IOW, for now, I'm sill keeping the HTTP GET in [a separate function](https://github.com/jackgene/easyracer/blob/47ba30a07511d9e087b234e2a49d6267ac4ee232/go-stdlib/main.go#L14-L36), because I feel like rolling this into the scenarios would make the scenarios too noisy, and doesn't really add to helping the reader understand the actual concurrency logic.

In the meantime, I'm going to look around for a good higher-level Go HTTP client to see if I can make the HTTP call a one/two liner. If I can do that, maybe it would be better to inline the whole thing.

**Comment by jackgene** [comment]
Hey James,
I think I'm happy with the Go implementation as it is, but am happy to remove the one DRY function if you want.

Here are the pros and cons of having a dedicated [httpText()](https://github.com/jackgene/easyracer/blob/45d19758ec61ff834a5f46a7165ee15d89e4674e/go-stdlib/main.go#L14-L36) function to make a single HTTP request, verify that it's a 200 response, and return the body as text, handling errors as appropriate:

The benefit is that each scenario only has code demonstrating how it is accomplishing its concurrency goal. It's not polluted by all of Go's verbose error handling when dealing with HTTP, and so it's easier to follow, while still demonstrating the use of Go routines and channels.

The drawback is that it makes Go seem deceptively simpler, and more expressive than it really is. It does not give readers a feel for what real Go code would look like.

Let me know what you think.

**Comment by jamesward** [comment]
Amazing work Jack! I'm merging it with the DRY http call code and will think more about how to handle this across all of the clients.
